src/ansys/pre_commit_hooks/add_license_headers.py

```diff
@@ -279,6 +283,53 @@ def check_exists(changed_headers, parser, values, proj, missing_headers, i):
     return changed_headers
 
 
+def add_hook_changes(before_hook, after_hook):
+    """
+    Add earlier hook changes to updated file with header.
+
+    Parameters
+    ----------
+    before_hook: str
+        Path to file before add-license-headers was run.
+    after_hook: str
+        Path to file after add-license-headers was run.
+    """
+    # Open files
+    before_file = open(before_hook, "r")
+    after_file = open(after_hook, "r")
+
+    # Compare the files before and after the hook ran
+    diff = difflib.ndiff(after_file.readlines(), before_file.readlines())
+
+    before_file.close()
+    after_file.close()
+
+    # Combine the changes from before the hook with the changes after the hook
+    with NamedTemporaryFile(mode="w", delete=False) as tmp:
+        while True:
+            try:
+                line = str(next(diff))
+                if "+" in line[0] and line[2:] == "\n":
+                    tmp.write(line[2:])
+                elif "-" in line[0]:
+                    tmp.write(line[2:])
+                elif "?" in line[0]:
+                    continue
+                elif " " in line[0]:
+                    tmp.write(line[2:])
+            except StopIteration:
+                break
```

Comment:
Why can't you loop on diff entries?

`` for entry in diff:``

This is much safer than a ``while True`` loop.

Also, I fail to understand these "very specific" cases you are considering here. Let's talk whenever you are around.

Reply:
I think I initially did `for entry in diff` but it would throw a StopIteration exception if the files were different sizes. 

In `diff = difflib.ndiff(after_file.readlines(), before_file.readlines())`, difflib.ndiff's output has +, -, or ? depending on the changes between the two files being compared. Whenever the line is the same between the two files, there is no symbol which is why one of the checks is " "